File: src/main/java/me/testaccount666/serversystem/commands/executables/teleportask/TeleportRequest.java
package me.testaccount666.serversystem.commands.executables.teleportask;

import lombok.Getter;
import lombok.Setter;
import me.testaccount666.serversystem.userdata.User;

public final class TeleportRequest {
    @Getter
    private final User _sender;
    @Getter
    private final User _receiver;
    private final long _timeout;
    @Getter
    private final boolean _teleportHere;
    @Setter
    @Getter
    private boolean _cancelled = false;
    @Setter
    @Getter
    private int _timerId;

    public TeleportRequest(User sender, User receiver, long timeout, boolean teleportHere) {
        _sender = sender;
        _receiver = receiver;
        _timeout = timeout;
        _teleportHere = teleportHere;
    }

    public boolean isExpired() {
        return System.currentTimeMillis() >= _timeout;
    }

}

File: src/main/kotlin/me/testaccount666/serversystem/commands/executables/teleportask/TeleportRequest.kt
package me.testaccount666.serversystem.commands.executables.teleportask

import me.testaccount666.serversystem.userdata.User

data class TeleportRequest(val sender: User, val receiver: User, private val _timeout: Long, val isTeleportHere: Boolean) {
    var isCancelled = false
    var timerId = 0

    val isExpired
        get() = System.currentTimeMillis() >= _timeout
}
